src/awkward/_v2/contents/content.py

```diff
@@ -245,14 +259,17 @@ def __getitem__(self, where):
                     )
                     allow_lazy = "copied"  # True, but also can be modified in-place
                 elif issubclass(where.dtype.type, (np.bool_, bool)):
-                    carry = ak._v2.index.Index64(np.nonzero(where.data.reshape(-1))[0])
+                    nplike = ak.nplike.of(where)
+                    where = nplike.nonzero(where.data.reshape(-1))[0]
```

Comment:
I think a better choice than `nplike = ak.nplike.of(where)` is `nplike = self.nplike`, which is to say, "regardless of whether the `where` array is on the CPU or GPU, move it to be on the same device as `self`." In fact, `where` could be a non-array, like a Python list, in which case `ak.nplike.of(where)` would return NumPy, but `self.nplike` could be CuPy. It will have to eventually be moved to the device `self` is on, so better to do that early.

Since this is only used once on the following line, `self.nplike` can just be used inline there.